protocol/adapter/protocol_whisper.go

```diff
@@ -102,40 +99,34 @@ func (w *ProtocolWhisperAdapter) Send(ctx context.Context, data []byte, options
 		return nil, err
 	}
 
-	if w.pfs != nil {
-		var (
-			encryptedData []byte
-			err           error
-		)
+	var newMessage *whisper.NewMessage
 
+	if w.publisher != nil {
 		privateKey := w.transport.KeysManager().PrivateKey()
 
+		var err error
+
 		// TODO: rethink this
 		if options.Recipient != nil {
-			encryptedData, err = w.pfs.BuildDirectMessage(
-				privateKey,
-				options.Recipient,
-				data,
-			)
+			newMessage, err = w.publisher.CreateDirectMessage(privateKey, options.Recipient, false, data)
 		} else {
-			encryptedData, err = w.pfs.BuildPublicMessage(privateKey, data)
+			_, filterErr := w.publisher.LoadFilter(&msgfilter.Chat{
+				ChatID: options.ChatName,
+			})
+			if filterErr != nil {
+				return nil, errors.Wrap(filterErr, "failed to load filter")
+			}
+
+			// TODO(adam): when wrap (the last argument) is important?
```

Comment:
Currently public messages are not wrapped (i.e have not bundle), when sending in public chats as it would be a breaking change. When we send a contact code, we send a public message but wrapped, as PFS enabled client are the only ones using it.

Reply:
That's helpful! Thanks.